### PR TITLE
chore: per-agent pool k8s versions and provisioning state

### PR DESCRIPTION
**Pull request by kkmsft**
- Add per agentpool level kubernetes version and provisioning state.
- Add calls to delete an entire vmss.

**Reason for Change**:

Required work towards multi agentpool operations.

**Issue Fixed**:

These will start getting used by AKS multi agent pool work initially and then used in aks-engine multi agentpool operations - like delete, scale, upgrade.

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
⚠️  PR does not follow commit message convention ⚠️

File: pkg/api/types.go [hunk 1]
```diff
@@ -486,6 +486,7 @@ type AgentPoolProfile struct {
 	DNSPrefix                           string               `json:"dnsPrefix,omitempty"`
 	OSType                              OSType               `json:"osType,omitempty"`
 	Ports                               []int                `json:"ports,omitempty"`
+	ProvisioningState                   ProvisioningState    `json:"provisioningState,omitempty"`
```
Comment (tariq1890) [inline pkg/api/types.go]: For external consumption?

Comment (CecileRobertMichon) [reply pkg/api/types.go]: there's already a provisioningState field in Properties, how is this different?

File: pkg/api/types.go [hunk 2]
```diff
@@ -503,6 +504,7 @@ type AgentPoolProfile struct {
 	PreprovisionExtension               *Extension           `json:"preProvisionExtension"`
 	Extensions                          []Extension          `json:"extensions"`
 	KubernetesConfig                    *KubernetesConfig    `json:"kubernetesConfig,omitempty"`
+	KubernetesVersion                   string               `json:"kubernetesVersion"`
```
Comment (CecileRobertMichon) [inline pkg/api/types.go]: why do we need this? do we expect heterogeneous k8s version support across agentpools?

File: pkg/api/types.go (hunk 1 above)
Comment (kkmsft) [reply pkg/api/types.go]: you mean - for user consumption in AKS - yes.

File: pkg/api/types.go (hunk 2 above)
Comment (jackfrancis) [reply pkg/api/types.go]: Just to be consistent with the pre-existing types I have a slight preference for `OrchestratorVersion` here.

**Comment by jackfrancis** [comment]
lgtm

**Pull request merged**